### PR TITLE
CtrlV.link image upload integration

### PR DESCRIPTION
**Pull request by lukyer**
Adding docs info about requested ctrlv.link/ctrlv.cz image uploader integration.
Discussion: https://github.com/Chatterino/chatterino2/discussions/3513

I'm open to any reordering you might prefer. I put it as a first item because this one doesn't require any authentication/accounts nor self-hosting.

### CODE REVIEW

**Comment by Felanbird** [comment]
Immediate issue I notice is since you don't use a direct link, [Chatterino/api](https://github.com/Chatterino/api/) cannot resolve your website for thumbnails, all of our existing examples "just work" because of direct links/opengraph tags. 
image

My opinion on the best approach for this would be just sending us the entire direct link, since the direct link includes the date and is already resolved by the api.
image


examples that just work
image
image
imgur
image
fourtf/i self host
image
gyazo as extra example
image

**Comment by lukyer** [comment]
@Felanbird ah, thanks! This functionality (opengraph thumbnails with smaller size) is implemented. I have just whitelisted Chatterino client. Please try it again now. It looks good to me.

**Comment by Felanbird** [comment]
> @Felanbird ah, thanks! This functionality (opengraph thumbnails with smaller size) is implemented. I have just whitelisted Chatterino client. Please try it again now. It looks good to me.

The noted issue has been resolved 👍

File: docs/Image Uploader.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ You can drag and drop images to Chatterino or paste them from clipboard to uploa
 By default, the image uploader is turned off. You can enable by checking the `Chatterino Settings -> External Tools -> Image Uploader -> Enable image uploader` option and configuring per one of the below options.
 
 Note to advanced users: This module sends multipart-form requests via POST method, so uploading via SFTP/FTP won't work.
-However, popular hosts like [imgur.com](https://imgur.com) and [s-ul.eu](https://s-ul.eu) are supported. Scroll down to see example cofiguration.
+However, popular hosts like [imgur.com](https://imgur.com), [s-ul.eu](https://s-ul.eu) and [ctrlv.link](https://ctrlv.link) are supported. Scroll down to see example cofiguration.
```
Comment (ALazyMeme) [inline docs/Image Uploader.md:7]: I'm not so sure about this edit. i.nuuls.com isn't included in this list, and it's a highly used website by a lot of users on Twitch.

File: docs/Image Uploader.md [hunk 2]
```diff
@@ -16,6 +16,16 @@ However, popular hosts like [imgur.com](https://imgur.com) and [s-ul.eu](https:/
...
+### ctrlv.link
+|Row|Description|
+|-|-|
+|Request URL|`https://ctrlv.link/upload.php`|
+|Form field|`img`|
+|Image link|`https://ctrlv.link/{link}`|
+|Deletion link|`https://ctrlv.link/delete.php?token={token}&id={link}`|
+
+You can also replace ctrlv.link with ctrlv.cz or ctrlv.sk to your language convenience.
+
```
Comment (ALazyMeme) [inline docs/Image Uploader.md:28]: New websites have been added to the bottom in past PRs. I think we should add this to the bottom as opposed to the top.

**Review by ALazyMeme — CHANGES_REQUESTED** [review]

**Comment by gempir** [comment]
Your website is missing something basic as a legal notice.
No privacy policy as far as I can see either.

The way you are using PHP makes me also think you are not using basic modern patterns like the [front controller pattern](https://en.wikipedia.org/wiki/Front_controller)

And since your software is closed source nobody can validate its security besides you. 

That makes me strongly dislike just adding your website to the Chatterino wiki, it's not supposed to be an ad space.

**Comment by pajlada** [comment]
The way I envision new image upload integrations to be added is through enough community members asking Chatterino contributors through the Discussions tag in the [main repo](https://github.com/Chatterino/chatterino2), or through our Discord, about how to use it with Chatterino.

Having all discussion created by the owner of the closed-source/commercial image uploader me and other contributors haven't heard of gives me strange vibes, which is why I'm closing this PR.

Feel free to continue to use this PR or the [discussion you created](https://github.com/Chatterino/chatterino2/discussions/3513) if you have any questions about this decision (or reach out on the Discord that you can find top-right of https://chatterino.com/)

**Comment by lukyer** [comment]
> Your website is missing something basic as a legal notice. No privacy policy as far as I can see either.

Privacy notice minimum required is in the footer. The reason why people use this well established service is mainly because it is easy to use and not bloated with nonsenses. 

> The way you are using PHP makes me also think you are not using basic modern patterns like the [front controller pattern](https://en.wikipedia.org/wiki/Front_controller)

Sorry but this is absolutely offtopic. I now what suits best this kind of service and technology. You may prefer to choose any design patterns by trends but it doesn't mean everyone does. 

> And since your software is closed source nobody can validate its security besides you.

If you want to validate security of the service, feel free to do another pentest. I'm happy to discuss any findings. We are doing blackbox pentests regularly. 

> That makes me strongly dislike just adding your website to the Chatterino wiki, it's not supposed to be an ad space.

I absolutely don't get your attitude here. I was asked by twitch users to add this integration. I'm making such open source integrations when multiple people are asking for it. This time I set up my very first twitch account to investigate it. I didn't know Chatterino SW before at all.


I already regret trying to contribute to this open source project.

**Comment by lukyer** [comment]
> The way I envision new image upload integrations to be added is through enough community members asking Chatterino contributors through the Discussions tag in the [main repo](https://github.com/Chatterino/chatterino2), or through our Discord, about how to use it with Chatterino.

Please mention it somewhere - I was looking for correct way how to contribute with these user requests. If I knew this I would ask requesters to start some discussion thread here. Maybe someone in the future will be in similar situation.

> Having all discussion created by the owner of the closed-source/commercial image uploader me and other contributors haven't heard of gives me strange vibes, which is why I'm closing this PR.

Let me cite this part of my yesterday's discussion about what Chatterino is:
"..jedná se o asi nejrozšířenější klient na twitch v zahraničí i v česku.."

> Feel free to continue to use this PR or the [discussion you created](https://github.com/Chatterino/chatterino2/discussions/3513) if you have any questions about this decision (or reach out on the Discord that you can find top-right of https://chatterino.com/)

Just wanted to justify for the record some misinformation posted here. I'm not interested in any more contributions here.

**Comment by gempir** [comment]
My comments about development decisions are off topic I agree, sorry. 

I just would like some more information how you handle customer data. 
Here are some pointers https://gdpr.eu/privacy-notice/

**Comment by subzero1177** [comment]
Hello, I was told to ask here about my configuration described in https://github.com/Chatterino/chatterino2/issues/3552
How can I set up the integration? Thanks in advance.

**Comment by LosFarmosCTL** [comment]
> Hello, I was told to ask here about my configuration described in [Chatterino/chatterino2#3552](https://github.com/Chatterino/chatterino2/issues/3552)
> How can I set up the integration? I see many links around in channels I'm in. So I'm curious how to set it up? Thanks in advance.

There is a wiki for chatterino with a dedicated page about [how to set custom image uploaders](https://wiki.chatterino.com/Image%20Uploader/). Also just scrolling through chatterino's settings for a maximum of 2 minutes, under "External tools" you would have found a form that exactly matches the screenshot you linked yourself, where you can just fill in the information from that screenshot.

And to reiterate, this is not a ctrlv support forum, if anything doesn't work with that configuration, no one here can do anything about that, ask the creator for help.

**Comment by subzero1177** [comment]
Hey, that's the missing bit! I combined those two instructions and It seems to work now, thanks a lot! 🥳

**Pull request closed without merging**